File: src/main/java/community/whatever/onembackendjava/entity/BlockedDomain.java
package community.whatever.onembackendjava.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

/**
 * 차단된 도메인 엔티티
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BlockedDomain {
    private Long id;
    private String domain;
    private Instant createdAt;
}

File: src/main/kotlin/community/whatever/onembackendkotlin/domain/BlockedDomain.kt
package community.whatever.onembackendkotlin.domain

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.net.URL
import java.util.UUID

@Table("BLOCKED_DOMAIN")
data class BlockedDomain(val domain: String, @Id val id: UUID) {
    constructor(url: String) : this(domain = URL(url).host, id = UUID.randomUUID())
}
